src/table/components/virtualized-table/hooks/use-infinite-scroll-data.ts

```diff
@@ -43,51 +43,143 @@ const createNewRow = (
   };
 };
 
+const isRowMissingData = (rows: Row[], x: number, y: number, width: number) => {
+  const targetRow = rows[y] as Row | undefined;
+
+  if (!targetRow) {
+    return true; // Row is not cached
+  }
+
+  for (let colIndex = x; colIndex < x + width; colIndex++) {
+    if (!targetRow[`col-${colIndex}`]) {
+      return true; // Column is not cached
+    }
+  }
```

Comment:
Yes exactly.

A row might look like this `{ col-0, col-1, col-5, col-8 }` in which case it would be awkward at best to find all missing cells and only fetch them.